public/js/selfoss-base.js

```diff
@@ -97,6 +118,75 @@ var selfoss = {
         
         return values;
     },
+
+
+    loggedin: false,
+
+
+    setSession: function() {
+        Cookies.set('onlineSession', 'true', {expires: 10,
+                                              path: window.location.pathname});
+        selfoss.loggedin = true;
+    },
+
+
+    clearSession: function() {
+        Cookies.remove('onlineSession', {path: window.location.pathname});
+        selfoss.loggedin = false;
+    },
+
+
+    hasSession: function() {
+        selfoss.loggedin = Cookies.get('onlineSession') == 'true';
+        return selfoss.loggedin;
+    },
+
+
+    login: function(e) {
+        $('#loginform').addClass('loading');
+        f = $('#loginform form');
+        $.ajax({
+            type: 'POST',
+            url: 'login',
+            dataType: 'json',
+            data: f.serialize(),
+            success: function(data) {
+                if (data.success) {
+                    $('#password').val('');
+                    selfoss.setSession();
+                    selfoss.ui.login();
+                    selfoss.ui.showMainUi();
+                    selfoss.initUi();
+                    selfoss.events.initHash();
+                } else {
+                    selfoss.events.setHash('login', false);
+                    selfoss.ui.showLogin(data.error);
+                }
+            },
+            complete: function(jqXHR, textStatus) {
+                $('#loginform').removeClass('loading');
+            }
+        });
+        e.preventDefault();
+    },
+
+
+    logout: function() {
+        selfoss.clearSession();
```

Comment:
Again, this is all done with offline in mind and is only a hint that there was an online session last time.